src/main/java/it/gov/pagopa/payhub/activities/activity/treasury/TreasuryOpiIngestionActivityImpl.java

```diff
@@ -26,50 +28,87 @@
 @Lazy
 @Component
 public class TreasuryOpiIngestionActivityImpl implements TreasuryOpiIngestionActivity {
-  private final IngestionFlowFileType ingestionflowFileType;
-  private final IngestionFlowFileDao ingestionFlowFileDao;
-  private final IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService;
+    private final IngestionFlowFileType ingestionflowFileType;
+    private final IngestionFlowFileDao ingestionFlowFileDao;
+    private final IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService;
+    private final TreasuryUnmarshallerService treasuryUnmarshallerService;
+
+
+    public TreasuryOpiIngestionActivityImpl(
+                                            IngestionFlowFileDao ingestionFlowFileDao,
+                                            IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService,
+                                            TreasuryUnmarshallerService treasuryUnmarshallerService) {
+        this.ingestionflowFileType = IngestionFlowFileType.OPI;
+        this.ingestionFlowFileDao = ingestionFlowFileDao;
+        this.ingestionFlowFileRetrieverService = ingestionFlowFileRetrieverService;
+        this.treasuryUnmarshallerService = treasuryUnmarshallerService;
+    }
 
 
-  public TreasuryOpiIngestionActivityImpl(IngestionFlowFileDao ingestionFlowFileDao,
-                                          IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService) {
-    this.ingestionflowFileType = IngestionFlowFileType.OPI;
-    this.ingestionFlowFileDao = ingestionFlowFileDao;
-    this.ingestionFlowFileRetrieverService = ingestionFlowFileRetrieverService;
-  }
+    @Override
+    public TreasuryIufResult processFile(Long ingestionFlowFileId) {
+        List<String> iufIuvList = new ArrayList<>();
```

Comment:
is this just a iuf list?
if build using stream API we don't need to initialize it
```suggestion
        List<String> iufList ;
```